Objects/dictobject.c

```diff
@@ -4165,14 +4165,22 @@ static PySequenceMethods dictkeys_as_sequence = {
 static PyObject*
 dictviews_sub(PyObject* self, PyObject *other)
 {
-    PyObject *result = PySet_New(self);
-    PyObject *tmp;
-    _Py_IDENTIFIER(difference_update);
-
-    if (result == NULL)
+    PyObject *left = self;
+    if (PyDictKeys_Check(self)) {
```

Comment:
Would it be worthwhile to move the repeated code into a function? Would not it add an overhead?